client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +49,64 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final RetentionConfig retentionConfig;
+
+    public enum RetentionConfig {
+        /**
+         * This {@link ReaderGroup} is not a subscriber and does not participate in Consumption Based Retention.
+         *
+         * See <a href="https://github.com/pravega/pravega/wiki/PDP-47:-Pravega-Streams:-Consumption-Based-Retention">Consumption Based Retention</a>
+         *
+         */
+        NONE(false, false),
+
+        /**
+         * This {@link ReaderGroup} is a subscriber and can retain stream data using Consumption Based Retention. This will configure the
+         * {@link ReaderGroup} to truncate the stream according to its read positions. These read positions have to be provided
+         * manually in the form of a {@link StreamCut}, see this method:
+         *
+         * {@link ReaderGroup#updateRetentionStreamCut(Map)}
+         *
+         * See <a href="https://github.com/pravega/pravega/wiki/PDP-47:-Pravega-Streams:-Consumption-Based-Retention">Consumption Based Retention</a>
+         *
+         */
+        TRUNCATE_AT_USER_STREAMCUT(true, false),
+
+        /**
+         * This {@link ReaderGroup} is a subscriber and can retain stream data using Consumption Based Retention. This will configure the
+         * {@link ReaderGroup} to truncate the stream using the read positions from the last {@link Checkpoint}.
+         *
+         * See <a href="https://github.com/pravega/pravega/wiki/PDP-47:-Pravega-Streams:-Consumption-Based-Retention">Consumption Based Retention</a>
+         *
+         */
+        TRUNCATE_AT_LAST_CHECKPOINT(true, true);
+
+        private final boolean isReaderGroupASubscriber;
+        private final boolean autoTruncateAtLastCheckpoint;
+
+        RetentionConfig(boolean isReaderGroupASubscriber, boolean autoTruncateAtLastCheckpoint) {
+            this.isReaderGroupASubscriber = isReaderGroupASubscriber;
+            this.autoTruncateAtLastCheckpoint = autoTruncateAtLastCheckpoint;
+        }
+    }
+
    public static class ReaderGroupConfigBuilder implements ObjectBuilder<ReaderGroupConfig> {
        private long groupRefreshTimeMillis = 3000; //default value
        private long automaticCheckpointIntervalMillis = 30000; //default value
        // maximum outstanding checkpoint request that is allowed at any given time.
        private int maxOutstandingCheckpointRequest = 3; //default value
+       private RetentionConfig readerGroupRetentionConfig = RetentionConfig.NONE;
```

Comment:
look at my previous message. i think the new retentionconfig is unnecessary and there is a cleaner way for users to be supplied its policy than using the enum in it.

Reply:
Based on what we had discussed and my understanding we are keeping the enum.